src/openrc/rc.c

```diff
@@ -473,12 +473,31 @@ handle_signal(int sig)
 	errno = serrno;
 }
 
+static void
+do_early_hostname(void)
+{
+	/* Set hostname if available */
+	char *buffer = NULL;
+	size_t len;
+
+	if (rc_getfile(RC_SYSCONFDIR "/hostname", &buffer, &len)) {
```

Comment:
The `hostname` tool from net-tools on Linux has a `-b` flag that will fall back to setting the hostname to `localhost` if the file does not exist or is empty; this would probably be a good idea to implement, as on Linux, the hostname could be an empty string otherwise.  `hostname` also allows for comment lines (line starts with `#`) in `/etc/hostname`, which are ignored; this could also be useful.